e2e-test/src/test/java/de/sovity/edc/extension/e2e/connector/config/ApiConfig.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2023 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *      sovity GmbH - init
+ */
+
+package de.sovity.edc.extension.e2e.connector.config;
+
+import java.net.URI;
+import java.util.Map;
+
+public record ApiConfig(String baseUrl, String name, String path, int port) implements EdcConfig {
```

Comment:
- Feedback: The undocumented Java Record + Code with branches was confusing. I needed a bit too long to understand what this class is.
- "baseUrl" should be renamed to "host", due to the contents of getUri().
- The parameter "name" should be renamed to apiGroup
- In my opinion, host + port + path should appear in that order, with "apiGroup" being first (refactoring the method should allow you to sort-parameters)
- In my opinion, this class should either have some documentation that points to that it's documentation for one api group, or be called ApiGroupConfig. As ApiConfig I expected it to get several ports

Reply:
- I would stick to `baseUrl` as the protocol is also passed in this variable
- renamed to `apiGroup`, like the change
- Changed order of parameters
- Renamed class to `EdcApiGroupConfig`, which more explicitly tells what is does